sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -0,0 +1,416 @@
+        /// <param name="partitionId">The identifier of the partition events are being received from.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(8, Level = EventLevel.Error, Message = "An exception occurred while receiving events for Event Hub: {0} (Partition Id: '{1}'). Error Message: '{2}'")]
+        public void EventReceiveError(string eventHubName,
+                                      string partitionId,
+                                      string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(8, eventHubName ?? String.Empty, partitionId ?? String.Empty, errorMessage ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that a client is closing, which may corespond to an <see cref="EventHubClient" />,
+        ///   <see cref="EventHubProducer" />, or <see cref="EventHubConsumer" />.
+        /// </summary>
+        ///
+        /// <param name="clientType">The type of client being closed.</param>
+        /// <param name="eventHubName">The name of the Event Hub associated with the client.</param>
+        /// <param name="clientId">An identifier to associate with the client.</param>
+        ///
+        [Event(9, Level = EventLevel.Verbose, Message = "Closing a {0} client (EventHub '{1}'; Identifier '{2}').")]
+        public void ClientCloseStart(Type clientType,
+                                     string eventHubName,
+                                     string clientId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(9, clientType.Name, eventHubName ?? String.Empty, clientId ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that a client has been closed, which may corespond to an <see cref="EventHubClient" />,
+        ///   <see cref="EventHubProducer" />, or <see cref="EventHubConsumer" />.
+        /// </summary>
+        ///
+        /// <param name="clientType">The type of client being closed.</param>
+        /// <param name="eventHubName">The name of the Event Hub associated with the client.</param>
+        /// <param name="clientId">An identifier to associate with the client.</param>
+        ///
+        [Event(10, Level = EventLevel.Verbose, Message = "A {0} client has been closed (EventHub '{1}'; Identifier '{2}').")]
+        public void ClientCloseComplete(Type clientType,
+                                        string eventHubName,
+                                        string clientId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(10, clientType.Name, eventHubName ?? String.Empty, clientId ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while receiving events.
```

Comment:
Summary is not accurate (exception was encountered while closing a client).

Reply:
Thank you.   Copy/paste all of this, so I probably mangled a few.